meta/autoid/autoid.go

```diff
@@ -912,6 +912,133 @@ func (alloc *allocator) alloc4Sequence(tableID int64) (min int64, max int64, rou
 	return newBase, newEnd, round, nil
 }
 
+// NewAllocatorFromTempTblInfo creates an in-memory allocator from a temporary table info.
+func NewAllocatorFromTempTblInfo(tblInfo *model.TableInfo) Allocator {
+	hasRowID := !tblInfo.PKIsHandle && !tblInfo.IsCommonHandle
+	hasAutoIncID := tblInfo.GetAutoIncrementColInfo() != nil
+	// Temporary tables don't support auto_random and sequence.
+	if hasRowID || hasAutoIncID {
+		return &inMemoryAllocator{
```

Comment:
the rowid and autoid should be separated? @tangenta

Reply:
It's fine because they are also not separated in non-temporary tables.